static/js/ducks/observingRun.js

```diff
@@ -0,0 +1,69 @@
+import messageHandler from 'baselayer/MessageHandler';
+
+import * as API from '../API';
+import store from '../store';
+import { REFRESH_SOURCE } from "./source";
+
+export const FETCH_OBSERVING_RUN = 'skyportal/FETCH_OBSERVING_RUN';
+export const FETCH_OBSERVING_RUN_OK = 'skyportal/FETCH_OBSERVING_RUN_OK';
+
+export const FETCH_ASSIGNMENT = 'skyportal/FETCH_ASSIGNMENT';
+export const FETCH_ASSIGNMENT_OK = 'skyportal/FETCH_ASSIGNMENT_OK';
+
+export const fetchObservingRun = (id) => (
+  API.GET(`/api/observing_run/${id}`, FETCH_OBSERVING_RUN)
+);
+
+export const fetchAssignment = (id) => (
+  API.GET(`/api/assignment/${id}`, FETCH_ASSIGNMENT)
+);
+
+// Websocket message handler
+messageHandler.add((actionType, payload, dispatch, getState) => {
+  const { observingRun } = getState();
+
+  /* TODO: Normalize state shape to eliminate the action processor below.
+  The implementation below re-fetches the entire run when one of the run's
+  sources is updated. With a normalized state tree, this (potentially large) GET
+  could be avoided in favor of a single update to the Source entity on the frontend.
+   */
+
+  if (actionType === REFRESH_SOURCE) {
```

Comment:
Why not "REFRESH_ASSIGNMENT"? You're not re-fetching the whole source here

Reply:
Because the changes made to a source from the assignment page should also propagate to the source page, and vice versa. So to keep the `SourceHandler.put` method from getting saturated with many different types of websocket pushes, I just handle the "REFRESH_SOURCE" method within the `observingRun.js` reducer.

Reply:
@dannygoldstein I think this would better be handled by a more specific WS message. The reason the followup request handler uses FETCH_SOURCE is because followup request data are fetched as part of the source data in GET /api/source/<source_id>, whereas assignments are fetched via their own unique endpoint. Just check whether the assignment was updated in SourceHandler.put and if so push a more specific message like REFRESH_ASSIGNMENT or something from there. In this case it's easy to make it more specific, and simpler.

Reply:
Will update the assignment reducer to look for REFRESH_ASSIGNMENT. But will have it push out REFRESH_SOURCE as well, as on the source page the logic is identical to that used for follow-up requests.

Reply:
Implemented REFRESH_OBSERVING_RUN